opensrp-app/src/main/java/org/smartregister/repository/AllSharedPreferences.java

```diff
@@ -51,11 +53,16 @@ public void updateANMUserName(String userName) {
     public String fetchRegisteredANM() {
         return preferences.getString(ANM_IDENTIFIER_PREFERENCE_KEY, "").trim();
     }
+    public void updateANMRole(String role) {
+        preferences.edit().putString(ANM_IDENTIFIER_ROLE, role).commit();
+    }
 
     public boolean isRegisteredANM(String userName) {
         return preferences.getStringSet(ANM_IDENTIFIER_SET_PREFERENCE_KEY, new HashSet<>()).contains(userName);
     }
-
+    public String fetchRegisteredRole() {
+        return preferences.getString(ANM_IDENTIFIER_ROLE, "").trim();
```

Comment:
Do we want to support multi-user login as well? If so we need to add the username suffix. Preferences without multi-user suffix get cleared on log out, meaning a fresh login will not work offline (for role dependent functionality)

Reply:
I think it might be best to capture/store the roles in the Account Manager instead, especially since the list of roles could be updated any time on the admin backend. Its more security as well. See [RemoteLoginTask.java#L119](https://github.com/opensrp/opensrp-client-core/blob/4853ae93c0ed542e1d7d9d3a66993993832b86fc/opensrp-app/src/main/java/org/smartregister/login/task/RemoteLoginTask.java#L119) and